benchmarks/bench_ml.py

```diff
@@ -301,36 +328,37 @@ def evaluate_optimizer(surrogate, model, dataset, n_calls, random_state):
     # below seed is necessary for processes which fork at the same time
     # so that random numbers generated in processes are different
     np.random.seed(random_state)
-    problem = MLBench(model, dataset)
+    problem = MLBench(model, dataset, random_state)
     space = problem.space
 
     # initialization
-    estimator = surrogate(random_state=random_state)
     dimensions_names = sorted(space)
     dimensions = [space[d][0] for d in dimensions_names]
-    solver = Optimizer(dimensions, estimator, random_state=random_state)
 
     trace = []
-    best_y = np.inf
-
-    # optimization loop
-    for i in range(n_calls):
-        point_list = solver.ask()
+    # iteration index, best objective; values shared between function calls.
+    shared_values = [0, np.inf]
 
+    def objective(point_list):
         # convert list of dimension values to dictionary
         point_dct = dict(zip(dimensions_names, point_list))
 
         # the result of "evaluate" is accuracy / r^2, which is the more the better
         objective_at_point = -problem.evaluate(point_dct)
 
-        if best_y > objective_at_point:
-            best_y = objective_at_point
+        if shared_values[-1] > objective_at_point:
+            shared_values[-1] = objective_at_point
+
+            # remember the point, objective pair
+        trace.append((point_dct, objective_at_point, shared_values[-1]))
```

Comment:
The `results` instance as returned from the minimize function provides all the information that you need. I would prefer this to be done outside the objective function.